### PR TITLE
Complete file format for targetFile

### PR DESCRIPTION
**Pull request by frankhuster**
and also dismisses magic number PMD errors in annotations (ie: mostly column widths)

### CODE REVIEW

File: checkstyle.xml [hunk 1]
```diff
@@ -163,6 +163,7 @@
             <property name="severity" value="info"/>
             <property name="tokens" value="NUM_DOUBLE,NUM_FLOAT,NUM_INT,NUM_LONG"/>
             <property name="ignoreNumbers" value="-1,0,1,2,60,24,7,31,32"/> <!-- 60,24 & 7 are used when calculating dates.  31 in hash. 32 is Magic Johnson's number so it gets a free pass -->
+            <property name="ignoreAnnotation" value="true" />
```
Comment (larubbio) [inline checkstyle.xml]: Will I no longer have to do this:

```java
   // checkstyle made me do it
    public static final String FALSE_STRING = "false";

    @Field
    @Column(allowsNull = FALSE_STRING, length = 100)
    @NotNull
    @Size(min = 1, max = 100)
    private String name;
```

File: outbound-dialer/src/main/java/org/motechproject/nms/outbounddialer/service/impl/TargetFileServiceImpl.java [hunk 2]
```diff
@@ -165,27 +168,203 @@ private File createTargetFileDirectory() {
     }
 
 
-    private void writeSubscriptionRow(Subscription subscription, OutputStreamWriter writer) throws IOException {
-        //subscription id
-        writer.write(subscription.getSubscriptionId());
+    private void writeSubscriptionRow(String fileIdentifier, Subscription subscription, String imiServiceId,
```
Comment (llavoie) [inline outbound-dialer/src/main/java/org/motechproject/nms/outbounddialer/service/impl/TargetFileServiceImpl.java]: Per #180, we need to add one more column to the file that tells IMI how the user was subscribed (IVR or MCTS_UPLOAD), which you can get from the Subscription.

File: kilkari/src/main/java/org/motechproject/nms/kilkari/domain/Subscription.java [hunk 3]
```diff
@@ -80,6 +79,15 @@ public void setDeactivationReason(DeactivationReason deactivationReason) {
         this.deactivationReason = deactivationReason;
     }
 
+    /**
+     * Returns the weekId corresponding to today's date.
+     * @return
+     */
+    @Ignore int todaysWeekId() {
```
Comment (koshalt) [inline kilkari/src/main/java/org/motechproject/nms/kilkari/domain/Subscription.java]: how does the @Ignore work here?

File: outbound-dialer/src/main/java/org/motechproject/nms/outbounddialer/domain/AuditRecord.java [hunk 4]
```diff
@@ -8,35 +8,62 @@
...
 @Entity(tableName = "nms_obd_audit_records")
 public class AuditRecord {
+    /**
+     * The identifier field is used in each targetFile csv row combined with the subscription id to form a uniquely
+     * identifying id for a specific OBD record
+     */
     @Field
     @UIDisplayable(position = 0)
-    private FileType type;
+    private String identifier;
```
Comment (koshalt) [inline outbound-dialer/src/main/java/org/motechproject/nms/outbounddialer/domain/AuditRecord.java]: Not sure if I understand the comments above. The identifier here is then combined with the subscription id to make the unique id? is this the batch id or request id that IMI wants?

Comment (koshalt) [inline outbound-dialer/src/main/java/org/motechproject/nms/outbounddialer/domain/AuditRecord.java]: I think I understand it now, looking at the code further down. Can we call this something else like fileId or batchId? Mostly a nitpick though

File: kilkari/src/main/java/org/motechproject/nms/kilkari/domain/Subscription.java (hunk 3 above)
Comment (larubbio) [inline kilkari/src/main/java/org/motechproject/nms/kilkari/domain/Subscription.java]: It tells MDS to not create a database field for that member.

**Comment by koshalt** [comment]
mostly minor comments. lgtm otherwise. I'll let you do the merge after you've had a chance to look at the comments

File: kilkari/src/main/java/org/motechproject/nms/kilkari/domain/Subscription.java (hunk 3 above)
Comment (koshalt) [inline kilkari/src/main/java/org/motechproject/nms/kilkari/domain/Subscription.java]: Ok, makes sense. Any reason why we wouldn't have an access modifier on the field? (public/private)

Comment (frankhuster) [inline kilkari/src/main/java/org/motechproject/nms/kilkari/domain/Subscription.java]: It's not a field it's a method!

**Pull request merged**